lang/src/org/partiql/lang/eval/visitors/SelectStarVisitorTransform.kt

```diff
@@ -0,0 +1,117 @@
+                null -> {
+                    val fromSourceAliases = extractAliases(rewritten.from)
+
+                    val newProjection =
+                        PartiqlAst.build {
+                            projectList(
+                                fromSourceAliases.map { aliases ->
+                                    // We are concatenating 3 lists here
+                                    listOf(createProjectAll(aliases.asAlias)) +
+                                        (aliases.atAlias?.let { listOf(createProjectExpr(it)) } ?: emptyList()) +
+                                        (aliases.byAlias?.let { listOf(createProjectExpr(it)) } ?: emptyList())
+                                }.flatten()
+                            )
+                        }
+                    return copyProjectionToSelect(rewritten, newProjection)
+                }
+                else -> {               // With group by
+                    val selectListItemsFromGroupBy = rewritten.group.keyList.keys.map {
+                        val asName = it.asAlias
+                            ?: errNoContext(
+                                "GroupByItem has no AS-alias--GroupByItemAliasVisitorTransform must be executed before SelectStarVisitorTransform",
+                                internal = true)
+
+                        // We need to take the unique name of each grouping field key only because we need to handle
+                        // the case when multiple grouping fields are assigned the same name (which is currently allowed)
+                        val uniqueNameMeta = asName.metas[UniqueNameMeta.TAG] as? UniqueNameMeta?
+                            ?: error("UniqueNameMeta not found--normally, this is added by GroupByItemAliasVisitorTransform")
+
+                        createProjectExpr(uniqueNameMeta.uniqueName, asName.text)
+                    }
+
+                    val groupNameItem = rewritten.group.groupAsAlias?.text.let {
+                        if (it != null) listOf(createProjectExpr(it)) else emptyList()
+                    }
+
+                    val newProjection = PartiqlAst.build { projectList(selectListItemsFromGroupBy + groupNameItem) }
+
+                    return copyProjectionToSelect(rewritten, newProjection)
+                }
+            }
+        }
+        return rewritten
+    }
+
+    private fun createProjectAll(name: String) =
+        PartiqlAst.build {
+            projectAll(id(name, caseSensitive(), unqualified(), emptyMetaContainer()))
+        }
+
+    private fun createProjectExpr(variableName: String, asAlias: String = variableName) =
+        PartiqlAst.build {
+            projectExpr(id(variableName, caseSensitive(), unqualified(), emptyMetaContainer()), asAlias)
+        }
+
+    private class FromSourceAliases(val asAlias: String, val atAlias: String?, val byAlias: String?)
+
+    private fun extractAliases(fromSource: PartiqlAst.FromSource): List<FromSourceAliases> =
```

Comment:
This follows what the previous rewriter did. Perhaps, we could use a PIG visitor to perform this like what is already used in EvaluatingCompiler ([L894](https://github.com/partiql/partiql-lang-kotlin/blob/36fe55b6457f0dc82c02da6975358c01dd13aa3b/lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt#L894)).

Reply:
I would encourage you to do that as part of this PR, but it's not *strictly* required.

Reply:
Changed to use PIG visitor.